File: iosApp/iosApp/Presentation/Main/CafeManagement/MenuGoods/MenuGoodsAction.swift
//
//  MenuGoodsAction.swift
//  ConCafe
//
//  Created by 홍희표 on 2026/03/11.
//

import Foundation

enum MenuGoodsAction {
    case clickBack
    case clickSearch
    case changeSearchQuery(String)
    case selectCollection(MenuGoodsUiState.CollectionTab)
    case selectCategory(String?)
    case toggleItemAvailability(String)
    case clickEditItem(String)
    case clickDeleteItem(String)
    case clickAddNewItem
    case dismissInfoMessage
}

File: composeApp/src/commonMain/kotlin/org/hhp227/concafe/presentation/main/cafemanagement/menugoods/MenuGoodsAction.kt
package org.hhp227.concafe.presentation.main.cafemanagement.menugoods

sealed interface MenuGoodsAction {
    data object ClickBack : MenuGoodsAction
    data object ClickSearch : MenuGoodsAction
    data class ChangeSearchQuery(val value: String) : MenuGoodsAction
    data class SelectCollection(val collection: MenuGoodsUiState.CollectionTab) : MenuGoodsAction
    data class SelectCategory(val categoryId: String?) : MenuGoodsAction
    data class ToggleItemAvailability(val itemId: String) : MenuGoodsAction
    data class ClickEditItem(val itemId: String) : MenuGoodsAction
    data class ClickDeleteItem(val itemId: String) : MenuGoodsAction
    data class ConfirmDeleteItem(val itemId: String) : MenuGoodsAction
    data object CancelDeleteItem : MenuGoodsAction
    data object ClickAddNewItem : MenuGoodsAction
    data object DismissInfoMessage : MenuGoodsAction
}
